Practicum/Pract. 08/Vector.cpp

```diff
@@ -0,0 +1,163 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include "Vector.h"
+#include <new>
+#include <exception>
+
+Vector::Vector() :Vector(4) {}
+
+
+Vector::Vector(unsigned capacityC): array(nullptr), capacity(capacityC), size(0)
+{
+	if (!capacityC)
+		throw "Bad capacity";
+
+	array = new(std::nothrow) String[capacityC];
+
+	if (!array)
+		throw std::bad_alloc();
+
+}
+
+Vector::Vector(const Vector& other)
+	:array(nullptr),capacity(other.capacity), size(other.size)
+{
+	copy(other);
+}
+
+Vector& Vector::operator=(const Vector& other)
+{
+	if (this != &other)
+	{
+		free();
+		copy(other);
+		
+	}
+	return *this;
+}
+
+Vector::~Vector()
+{
+	free();
+}
+
+
+unsigned Vector::getCapacity() const
+{
+	return this->capacity;
+}
+
+void Vector::push_back(const String& str)
+{
+	if (size == capacity)
+		resize(capacity * 2);
+
+	array[size++] = str;
+}
+
+void Vector::pop_back()
+{
+	if (!size)
+		return;
+
+	if (size <= capacity / 4)
+		resize(capacity / 4);
+
+	size--;	
+}
+
+void Vector::insert(unsigned index, const String& str)
+{
+	if (index >= size)
+		throw std::invalid_argument("bad index");
+	 
+	if (size + 1 == capacity)

```

Comment:
In the other functions for adding items you were just checking size == capacity.
Please be consistent with yourself